File: src/main/java/com/ject/studytrip/trip/domain/factory/TripReportStudyLogFactory.java
package com.ject.studytrip.trip.domain.factory;

import com.ject.studytrip.studylog.domain.model.StudyLog;
import com.ject.studytrip.trip.domain.model.TripReport;
import com.ject.studytrip.trip.domain.model.TripReportStudyLog;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class TripReportStudyLogFactory {
    public static TripReportStudyLog create(TripReport tripReport, StudyLog studyLog) {
        return TripReportStudyLog.of(tripReport, studyLog);
    }
}

File: src/main/kotlin/com/ject/studytrip/trip/domain/factory/TripReportStudyLogFactory.kt
package com.ject.studytrip.trip.domain.factory

import com.ject.studytrip.studylog.domain.model.StudyLog
import com.ject.studytrip.trip.domain.model.TripReport
import com.ject.studytrip.trip.domain.model.TripReportStudyLog

object TripReportStudyLogFactory {
    @JvmStatic
    fun create(
        tripReport: TripReport,
        studyLog: StudyLog,
    ): TripReportStudyLog = TripReportStudyLog.of(tripReport, studyLog)
}
